fsm.go

```diff
@@ -135,6 +144,18 @@ func (fsm *fsm) State() (consensus.State, error) {
 	return dupState(fsm.state)
 }
 
+func (fsm *fsm) updateSubscribers(st consensus.State) {
+	fsm.chMux.Lock()
+	defer fsm.chMux.Unlock()
+	if fsm.subscriberCh != nil {
+		select {
+		case fsm.subscriberCh <- st:
+		default:
+			logger.Error("subscriber channel is full. Discarding state!")
```

Comment:
In this case, does the user care about *every* individual state? or just the resultant state? 

If the former, Then we definitely should not be throwing away states. If the latter, then we should probably not buffer the channel, and have some extra buffer that holds just the latest state and sends it when it can.

I use a similar pattern here in bitswap: https://github.com/ipfs/go-ipfs/blob/master/exchange/bitswap/workers.go#L122

to handle buffering outgoing provider records without blocking or dropping anything.